store/cachekv/internal/memiterator.go

```diff
@@ -11,20 +11,18 @@ import (
 var _ types.Iterator = (*memIterator)(nil)
 
 // memIterator iterates over iterKVCache items.
-// if key is nil, means it was deleted.
+// if value is nil, means it was deleted.
 // Implements Iterator.
 type memIterator struct {
 	iter btree.IterG[item]
 
 	start     []byte
 	end       []byte
 	ascending bool
-	lastKey   []byte
-	deleted   map[string]struct{}
```

Comment:
Removal of this field allows us to close the other issue/item re deleted field, right?

Reply:
which one is that?

Reply:
https://github.com/cosmos/cosmos-sdk/issues/14376 which this PR is closing (my bad) 😆